BitmarkRegistry/Controllers/Main/Properties/BitmarkDetailViewController.swift

```diff
@@ -98,10 +102,48 @@ class BitmarkDetailViewController: UIViewController {
     sender.image = UIImage(named: actionMenuView.isHidden ? "More Actions-close" : "More Actions-open")
   }
 
+  @objc func closeActionMenu(_ sender: UIGestureRecognizer) {
+    guard !actionMenuView.isHidden else { return }
+    actionMenuView.isHidden = true
+    menuBarButton.image = UIImage(named: "More Actions-close")
+  }
+
   @objc func tapToCopyId(_ sender: UIButton) {
     UIPasteboard.general.string = bitmark.id
     copiedToClipboardNotifier.showIn(period: 1.2)
   }
+
+  @objc func tapToDownload(_ sender: UIButton) {
+    activityIndicator.startAnimating()
+    assetFileService.getSenderAccountNumber { [weak self] (senderAccountNumber, error) in
+      guard let self = self else { return }
+      if let error = error {
+        DispatchQueue.main.async {
+          self.activityIndicator.stopAnimating()
+          self.showErrorAlert(message: Constant.Error.downloadAsset)
+        }
+        ErrorReporting.report(error: error)
+        return
+      }
+
+      self.assetFileService.downloadFileFromCourierServer(senderAccountNumber: senderAccountNumber!, completion: { (downloadedFileURL, error) in
+        DispatchQueue.main.async {
+          self.activityIndicator.stopAnimating()
```

Comment:
Retain cycle here. There will be a chance user leaves this screen before download process finishes. So this ViewController cannot be disposed.

Reply:
I have just updated with another code function flow. I think it will remove the retain cycle here too. Thanks